engine/access/rpc/backend/backend_scripts.go

```diff
@@ -81,10 +84,45 @@ func (b *backendScripts) executeScriptOnExecutionNode(
 		Arguments: arguments,
 	}
 
-	execResp, err := b.executionRPC.ExecuteScriptAtBlockID(ctx, &execReq)
+	// if an executionRPC is provided, send the script to that execution node
+	if b.staticExecutionRPC != nil {
+		execResp, err := b.staticExecutionRPC.ExecuteScriptAtBlockID(ctx, &execReq)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to execute the script on the execution node: %v", err)
+		}
+		return execResp.GetValue(), nil
+	}
+
+	// find the execution nodes which have executed the block earlier and provided an execution receipt for it
+	execNodeIDs, err := executionNodesForBlockID(blockID, b.executionReceipts, b.state)
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to execute the script on the execution node: %v", err)
+		return nil, err
+	}
+
+	var errors error
+	// choose upto 3 random execution node to try
+	execNodes := execNodeIDs.Sample(3)
+
+	// try to execute the script on one of the three execution nodes
+	for _, execNode := range execNodes {
+		result, err := b.tryExecuteScript(ctx, execNode, execReq)
+		if err == nil {
+			return result, nil
+		}
+		multierror.Append(errors, err)
 	}
+	return nil, errors
```

Comment:
```suggestion
	var errors *multierror.Error
	// choose upto 3 random execution node to try
	execNodes := execNodeIDs.Sample(3)

	// try to execute the script on one of the three execution nodes
	for _, execNode := range execNodes {
		result, err := b.tryExecuteScript(ctx, execNode, execReq)
		if err == nil {
			return result, nil
		}
		errors = multierror.Append(errors, err)
	}
	return nil, errors.ErrorOrNil()
```

I don't think `Append` will modify the input, need to use the return value.